### PR TITLE
Add push-to-app-catalog github action

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/19941

This adds a reusable github action for building apps using ABS and pushing to a github repo.

I've been testing it with a test [catalog]( https://github.com/rossf7/example-catalog) and a fork of our [hello-world-app](https://github.com/rossf7/hello-world-app/blob/master/.github/workflows/push-to-app-catalog.yaml) and here is an example [run](https://github.com/rossf7/hello-world-app/runs/4940497070?check_suite_focus=true)
 
```yaml
name: 'Push to App Catalog'

on:
  push:
    tags:
      - 'v*'

jobs:
  push_to_app_catalog:
    uses: giantswarm/app-build-suite/.github/workflows/push-to-app-catalog.yaml@github-action
    with:
      app_catalog: example-catalog
      chart: hello-world-app
      organization: rossf7
    secrets:
      envPAT: ${{ secrets.PAT }}
```

### CODE REVIEW

File: .github/workflows/push-to-app-catalog.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          smoke_test: "${binary} --version"
+          tarball_binary_path: "*/${binary}_linux_amd64"
+          version: "v4.17.2"
+
+      - name: Set version from Chart.yaml
+        id: chart-version
+        run:  echo "::set-output name=version::$(yq e '.version' helm/${{ inputs.chart }}/Chart.yaml)"
+
+      - name: Commit changes to app catalog repository
+        uses: stefanzweifel/git-auto-commit-action@v4
```
Comment (rossf7) [inline .github/workflows/push-to-app-catalog.yaml:67]: This action works well but maybe we should create our own action for this?

https://github.com/stefanzweifel/git-auto-commit-action

File: .github/workflows/push-to-app-catalog.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        uses: actions/checkout@v2
+        with:
+          path: app_catalog
+          repository: ${{ inputs.organization }}/${{ inputs.app_catalog }}
+          token: ${{ secrets.envPAT }}
+
+      - name: Execute app-build-suite
+        uses: docker://quay.io/giantswarm/app-build-suite:1.0.4
+        with:
+          args: --chart-dir helm/${{ inputs.chart }} --catalog-base-url "https://${{ inputs.organization }}.github.io/${{ inputs.app_catalog }}/" --destination app_catalog --keep-chart-changes
```
Comment (rossf7) [inline .github/workflows/push-to-app-catalog.yaml:41]: We keep the chart changes to get the version from Chart.yaml

File: .github/workflows/push-to-app-catalog.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+      - name: Set version from Chart.yaml
+        id: chart-version
+        run:  echo "::set-output name=version::$(yq e '.version' helm/${{ inputs.chart }}/Chart.yaml)"
+
+      - name: Commit changes to app catalog repository
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          branch: main
+          commit_message: add ${{ inputs.chart }}-${{ steps.chart-version.outputs.version }}
```
Comment (rossf7) [inline .github/workflows/push-to-app-catalog.yaml:70]: The version is used for the commit message.

**Review by piontec — APPROVED** [review]
LGTM! As for the external action... I think we should treat it as any other library - we rely on external code heavily and I don't think this use case is different than an open source go library. TL;DR: if it works, let's keep it

**Review by MarcelMue — APPROVED** [review]
LGTM

**Comment by rossf7** [comment]
> As for the external action... I think we should treat it as any other library - we rely on external code heavily and I don't think this use case is different than an open source go library. TL;DR: if it works, let's keep it

@piontec Yes makes sense I just wanted to check.

I also made a small change to add `--replace-chart-version-with-git` so apps without an ABS config file have the correct version.

We can link to the docs so users can extend the config but I think it's good to have this as a default.

**Pull request merged**